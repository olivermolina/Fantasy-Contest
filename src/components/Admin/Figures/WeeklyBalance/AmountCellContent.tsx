import Box from '@mui/material/Box';
import Link from '@mui/material/Link';
import React from 'react';

export default function AmountCellContent(props: {
  amount: number;
  onClick?: () => void;
}) {
  if (props.amount === 0 && !props.onClick) {
    return (
      <Box>
        <span>{props.amount.toFixed(2)}</span>
      </Box>
    );
  }
  if (!props.onClick) {
    return (
      <Box
        sx={{
          color:
            props.amount !== 0 ? (props.amount > 0 ? 'green' : 'red') : 'black',
        }}
      >
        <span>{props.amount?.toFixed(2)}</span>
      </Box>
    );
  }

  return (
    <Link
      component="button"
      underline={'hover'}
      sx={{
        color:
          props.amount !== 0 ? (props.amount > 0 ? 'green' : 'red') : 'black',
      }}
      onClick={props.onClick}
    >
      <span>{props.amount?.toFixed(2)}</span>
    </Link>
  );
}
