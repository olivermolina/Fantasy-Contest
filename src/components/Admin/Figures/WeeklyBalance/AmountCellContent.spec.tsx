import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AmountCellContent from './AmountCellContent';

describe('AmountCellContent', () => {
  it('should render correctly', () => {
    const props = {
      amount: 110,
      onClick: jest.fn(),
    };

    const { container } = render(<AmountCellContent {...props} />);
    expect(container).toMatchSnapshot();
  });
});
