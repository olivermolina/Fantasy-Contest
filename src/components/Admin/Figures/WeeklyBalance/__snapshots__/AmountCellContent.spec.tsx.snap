// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountCellContent should render correctly 1`] = `
<div>
  <button
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover MuiLink-button css-1akz8jj-MuiTypography-root-MuiLink-root"
  >
    <span>
      110.00
    </span>
  </button>
</div>
`;
